File: src/main/java/org/gga/graph/search/dfs/AbstractDfsVisitor.java
package org.gga.graph.search.dfs;

import org.gga.graph.Graph;
import org.gga.graph.Edge;

/**
 * @author mike
 */
public class AbstractDfsVisitor implements DfsVisitor {
    public void initializeVertex(int v, Graph graph) {
    }

    public void startVertex(int v, Graph graph) {
    }

    public void discoverVertex(int v, Graph graph) {
    }

    public void examineEdge(Edge e, Graph graph) {
    }

    public void treeEdge(Edge e, Graph graph) {
    }

    public void backEdge(Edge e, Graph graph) {
    }

    public void forwardOrCrossEdge(Edge e, Graph graph) {
    }

    public void finishVertex(int v, Graph graph) {
    }
}

File: src/main/scala/org/gga/graph/search/dfs/AbstractDfsVisitor.scala
package org.gga.graph.search.dfs

import org.gga.graph.Edge
import org.gga.graph.Graph

/**
 * @author mike
 */
trait AbstractDfsVisitor extends DfsVisitor {
  override def initializeVertex(v: Int, graph: Graph) {
  }

  override def startVertex(v: Int, graph: Graph) {
  }

  override def discoverVertex(v: Int, graph: Graph) {
  }

  override def examineEdge(e: Edge, graph: Graph) {
  }

  override def treeEdge(e: Edge, graph: Graph) {
  }

  override def backEdge(e: Edge, graph: Graph) {
  }

  override def forwardOrCrossEdge(e: Edge, graph: Graph) {
  }

  override def finishVertex(v: Int, graph: Graph) {
  }
}